### PR TITLE
Sign in, Sign up

### PR DESCRIPTION
**Pull request by Nxthien**

### CODE REVIEW

File: lib_manager/config/routes.rb [hunk 1]
```diff
@@ -1,4 +1,12 @@
...
+  get 'sessions/new'
+
   root "pages#show", page: "home"
   get "/pages/*page", to: "pages#show"
+  get "signup", to: "users#new"
+  get "/login", to: "sessions#new"
+  post "/login", to: "sessions#create"
+  delete "/logout", to: "sessions#destroy"
+
+  resources :users, expert: [:new, :update, :edit, :index]
```
Comment (dungvuit) [inline lib_manager/config/routes.rb]: chia block

File: lib_manager/app/models/user.rb [hunk 2]
```diff
@@ -16,4 +19,47 @@ class User < ApplicationRecord
...
+
+  class << self
+    def new_token
+      SecureRandom.urlsafe_base64
+    end
+
+    def digest string
+      cost = ActiveModel::SecurePassword.min_cost ?
+        BCrypt::Engine::MIN_COST : BCrypt::Engine.cost
+        BCrypt::Password.create string, cost: cost
```
Comment (hellovietnam93) [inline lib_manager/app/models/user.rb]: ident

File: lib_manager/app/views/shared/_error_messages.html.erb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<div id="errors_panel">
+  <% if @user.errors.any? %>
+    <div id="error_explanation">
+      <div class="alert alert-danger">
+        <%= t "views.user.signup.errors_contain" %>
+          <%= pluralize @user.errors.count, "error" %>.
+      </div>
+      <ul>
+      <% @user.errors.full_messages.each do |msg| %>
```
Comment (hellovietnam93) [inline lib_manager/app/views/shared/_error_messages.html.erb]: ident

File: lib_manager/app/views/shared/_error_messages.html.erb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<div id="errors_panel">
+  <% if @user.errors.any? %>
+    <div id="error_explanation">
+      <div class="alert alert-danger">
+        <%= t "views.user.signup.errors_contain" %>
+          <%= pluralize @user.errors.count, "error" %>.
```
Comment (hellovietnam93) [inline lib_manager/app/views/shared/_error_messages.html.erb]: I18n cho erro

File: lib_manager/app/views/users/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,31 @@
+<% provide :title, t("views.user.signup.sign_up") %>
+<div id="new_form" class="user-form">
+  <h1>Sign up</h1>
```
Comment (hellovietnam93) [inline lib_manager/app/views/users/new.html.erb]: I18n

File: lib_manager/config/routes.rb (hunk 1 above)
Comment (hellovietnam93) [inline lib_manager/config/routes.rb]: expert laf sao?

File: lib_manager/app/views/sessions/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,26 @@
+<% provide :title, t("views.user.signins.login") %>
+<div id="log-in" class="user-form">
+  <h1>Log in</h1>
```
Comment (hellovietnam93) [inline lib_manager/app/views/sessions/new.html.erb]: I18n

File: lib_manager/app/views/sessions/new.html.erb [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+        <%= f.label :remember_me, class: "checkbox inline" do %>
+        <%= f.check_box :remember_me %>
+          <span>Remember me on this computer</span>
+        <% end %>
+
+        <%= f.submit t("views.user.signins.login"), class: "btn btn-primary" %>
+      <% end %>
+
+      <p>New user? <%= link_to t("views.user.signins.sign-up"), signup_path %></p>
```
Comment (hellovietnam93) [inline lib_manager/app/views/sessions/new.html.erb]: I18n

**Pull request merged**